command/agent/job_endpoint_test.go

```diff
@@ -1601,6 +1607,43 @@ func TestHTTP_JobVersions(t *testing.T) {
 			t.Fatalf("bad %v", v)
 		}
 
+		if len(vResp.Diffs) != 1 {
+			t.Fatalf("bad %v", vResp)
+		} else {
+			for _, jobDiff := range vResp.Diffs {
+				for _, taskGroupDiff := range jobDiff.TaskGroups {
+					for _, taskDiff := range taskGroupDiff.Tasks {
+						if len(taskDiff.Objects) == 0 {
+							continue
+						}
+						firstObject := taskDiff.Objects[0]
+						if firstObject.Type != "Added" {
+							t.Errorf("Expected first object type to be 'Added', got '%s'", firstObject.Type)
+							return
+						}
+						for _, field := range firstObject.Fields {
+							if field.Name == "Name" {
+								if field.New != "updated date test" {
+									t.Errorf("Expected new 'Name' to be 'updated date test', got '%s'", field.New)
+								}
+							}
+						}
+						for _, argObject := range firstObject.Objects {
+							if argObject.Name == "Args" {
+								for _, argField := range argObject.Fields {
+									if argField.Name == "Args" {
+										if argField.New != "--utc" {
+											t.Errorf("Expected new 'Args' to be '--utc', got '%s'", argField.New)
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
```

Comment:
There's a test suite in `nomad/structs/diff_test.go` that's supposed to cover `Diff`. Doing it here is testing far away from the point where the method is defined.

Reply:
Oh wow, that file is a lot more straightforward looking. Removing this and migrating to it now.